### DISCUSSION TITLE
[Docker] couldn't read record

### DISCUSSION THREAD

**Discussion by Pierre-Lannoy** [hubblo-org/scaphandre#395] [General]
## Bug description

Hello!
On a fresh new install, docker scaphandre returns errors (and 0 watts) with the following error: `Could'nt read record from /sys/class/powercap/intel-rapl:0/energy_uj, error was: Os { code: 2, kind: NotFound, message: "No such file or directory" }`.

## To Reproduce

On a fresh Debian install just do:

```
docker run -v /sys/class/powercap:/sys/class/powercap -v /proc:/proc -ti hubblo/scaphandre stdout
```

## Expected behavior

It should work, I think.

## Screenshots

*none*

## Environment
- Linux Debian 12.2.0-14
- Kernel version 6.1.0-18-amd64

## Additional context

I've executed `init.sh` script and everyone can read `/sys/class/powercap/intel-rapl:0/energy_uj`.

**Comment by zhangyi921** [comment]
I have the same issue!

**Comment by vbmade2000** [comment]
@Pierre-Lannoy  @bpetit 

I was facing same issue.
It is possibly because the container is unable to access the directories/files that require root privileges. To resolve the issue, you can use privileged containers as shown below. I can successfully run the command.

`docker run --privileged -v /sys/class/powercap:/sys/class/powercap -v /proc:/proc -ti hubblo/scaphandre stdout -t 15`

**Comment by bpetit** [comment]
Hi,

Yes, this is known. Adequate permissions have to be set on /sys/class/powercap files. The init.sh script does it, but is only there for testing purposes. In the beginning of scaphandre this was not an issue as those files were accessible with unprivileged user. A patch in the kernel changed that, which version is more and more present in all linux distributions, so now we need privileges to access them. This plus many other reasons : #135 #106 make running scaphandre (in docker but not only) needing privileges. --privileged will fix this in docker as @vbmade2000 said.
